solution1/solution1.go

```diff
@@ -0,0 +1,180 @@
+package main
+
+import (
+	"bytes"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"os/user"
+	"path"
+	"strings"
+)
+
+func init() {
+	log.SetFlags(log.Ldate | log.Ltime | log.LUTC | log.Lshortfile)
+	log.SetPrefix("I ")
+}
+
+type Config struct {
+	K8sServer  string
+	K8sCrtFile string
+	K8sKeyFile string
+	K8sCAFile  string
+}
+
+func main() {
+	config := parseFlags()
+
+	err := mainError(config)
+	if err != nil {
+		log.SetPrefix("E ")
+		log.Printf("%s", err)
+		os.Exit(1)
+	}
+}
+
+func parseFlags() Config {
+	var config Config
+
+	var homedir string
+	{
+		u, err := user.Current()
+		if err != nil {
+			homedir = os.Getenv("HOME")
+		} else {
+			homedir = u.HomeDir
+		}
+
+	}
+
+	var serverDefault string
+	{
+		out, err := exec.Command("minikube", "ip").Output()
+		if err == nil {
+			minikubeIP := strings.TrimSpace(string(out))
+			serverDefault = "https://" + string(minikubeIP) + ":8443"
+		}
+	}
+
+	flag.StringVar(&config.K8sServer, "kubernetes.server", serverDefault, "Kubernetes API server address.")
+	flag.StringVar(&config.K8sCrtFile, "kubernetes.crt", path.Join(homedir, ".minikube/apiserver.crt"), "Kubernetes certificate file path.")
+	flag.StringVar(&config.K8sKeyFile, "kubernetes.key", path.Join(homedir, ".minikube/apiserver.key"), "Kubernetes key file path.")
+	flag.StringVar(&config.K8sCAFile, "kubernetes.ca", path.Join(homedir, ".minikube/ca.crt"), "Kubernetes CA file path.")
+	flag.Parse()
+
+	return config
+}
+
+func mainError(config Config) error {
+	k8sClient, err := newHttpClient(config)
+	if err != nil {
+		return err
+	}
+
+	// Create Custom Resource Definition.
+	{
+		log.Printf("creating custom resource")
+
+		res, err := k8sClient.Post(
+			config.K8sServer+"/apis/apiextensions.k8s.io/v1beta1/customresourcedefinitions",
+			"application/json",
+			strings.NewReader(crdJson),
+		)
+		if err != nil {
+			return err
+		}
+		defer res.Body.Close()
+
+		switch res.StatusCode {
+		case http.StatusConflict:
+			m := make(map[string]interface{})
+			err := json.NewDecoder(res.Body).Decode(&m)
+			return fmt.Errorf("creating custom resource: decoding body=%s", readerToString(res.Body))
```

Comment:
Can the log say it's creating the resource definition?  I can see newbies getting confused between CRDs and CROs at the start.

Reply:
I think it is ok to call it `custom resource` and `custom object`. (1) it's interchangeable with TPR stuff. (2) Below you have a log for waiting for custom resource to be ready, which is something different than waiting for custom resource definition to be ready. Custom resource is something Kubernetes creates as a result of custom resource definition object, and we should make that clear. Do you agree?

Reply:
Yes, I like custom resource and custom object. Nice that it's interchangeable with TPRs.